### ISSUE TITLE
Allow scrolling transient popup

### ISSUE THREAD

**Issue by aplaice** [magit/transient#7]
Would it be possible to have an option to scroll the transient "popup"? Currently, when the transient is long, the emacs frame is small or if one is using a large font (none of which is a rare situation), not all of the transient is visible (see attached image — the "log" and reflog" commands are all invisible). There appears to be no option to scroll the transient, to make the bottom section visible. Even dragging the scroll-bar has no effect. This is rather inconvenient, especially since the "action" (suffix?) commands are at the bottom.

Would it be possible to add a set of keybindings to make scrolling possible?

Some possibilities might be `SPC` and `S-SPC`, `C-v` and `M-v`, `next` and `prior` (i.e. `PgDown` and `PgUp`). Of these, `C-v` seems to be taken, so that pair is probably not a good idea, but the others appear to be free.

In addition, allowing scrolling with the mouse (`<mouse-[45]>`), if at all possible, would also be useful.

Thank you, for yet another, otherwise wonderful package!

![magit_transient_with_small_emacs_frame](https://user-images.githubusercontent.com/16461345/52900323-dcefe000-31f4-11e9-96b4-70a2a2f4f896.png)

**Comment by tarsius** [comment]
> Would it be possible to add a set of keybindings to make scrolling possible?

Not on our side. This is something that would have to be implemented in the `lv` library, which is used to *display* the transient.

**Comment by tarsius** [comment]
Since Hydra has been using `lv` for a long time there probably are past discussions about this issue. If you find any, then please point me there.

**Comment by aplaice** [comment]
Thanks for the fast reply!

I haven't been able to find any discussion about scrolling the popup. (Searching for `scroll` in `hydra`'s issues, only brings up (closed) issues where somebody wanted to scroll the "initial"/"main" buffer. I can't think of any other relevant keywords.)

However, testing with `hydra` itself seems to show that scrolling the "popup" works fine. For instance, one can scroll with the mouse-wheel (`<mouse-[45]>`), by dragging the scroll-bar or using `C-M-v` and `C-S-M-v`. 

(As I hadn't used `hydra` before, I carried out testing using the [example in the readme](https://github.com/abo-abo/hydra#the-impressive-looking-one), adapting it slightly by adding extra new lines in the middle docstring to make sure that it overflowed.)

Interestingly, if one focuses `hydra`'s popup by clicking on it (using `C-x o` doesn't work) then, even though the cursor isn't displayed, all the normal emacs keybindings work, so one can even scroll up and down with `<next>`/`<prior>` or `C-v`/`M-v`, provided that they haven't been set in the hydra. (As an aside, all of Emacs's normal keybindings working has the upside that one can search with `C-s` or select text with the keyboard or the mouse — again provided that the relevant keys haven't been set within the hydra — but the downside that one can inadvertently change the contents of the "popup"...)

I'm not yet sure if this is something extra that `hydra` provides or if it's possible with `lv` itself.

**Comment by aplaice** [comment]
Actually, all this apparently works with `lv` itself, if one naïvely creates an lv window with `(lv-message "some long multiline text")`.

OTOH `transient` rejecting most keybindings that weren't specifically bound by it is probably a good idea to avoid confusing the user and, potentially, data loss.

**Comment by tarsius** [comment]
Did you test with a "pink" hydra? I took a quick look and didn't see anything that explicitly enabled the keys/commands you mentioned. But if this is possible, then we should be able to get the same behavior in transient by "whitelisting" certain non-suffix commands.

**Comment by aplaice** [comment]
Yes, it was a pink hydra, so it was just a case of allowing foreign keys. (Sorry for not stating that explicitly, above!)

**Comment by tarsius** [comment]
Scrolling is possible using either the mouse wheel or they keys that usually scroll the current window.

**Comment by aplaice** [comment]
Thanks for implementing this! The way you keep everything well-documented is really inspiring!

(Also, sorry for taking up your time with a relatively niche issue. If I had realised that it was relatively straightforward, I'd have tried doing this myself, but my earlier first attempts had encountered a cryptic error message (unfortunately I don't remember what it was) and I gave up.)

There seem to be three issues — I've opened a pull request to fix them, hopefully without breaking anything else:

1. I think there's a typo in the definition of `transient-scroll-down`. 

2. `scroll-up-command` and `scroll-down-command` aren't bound in the transient, so the `remap scroll-(up|down)-command` has no actual effect and hence there is no keybinding for scrolling, by default, contrary to the statement in the manual (though setting one with `define-key` does work).

3. (Not really an issue, more of an enhancement.) `transient-scroll-(up|down)` can take an argument, but an end user cannot actually make use of this, without overriding or advising `transient--redisplay` (which is messy). If user were to define a new scroll function wrapping around, say, `(transient-scroll-up 1)` and bind that in one of the transient maps (and "bind" the function in the `transient-predicate-map` to `transient--do-stay`), it wouldn't work because the function wouldn't be exempted from causing `transient--show` to be called in `transient--redisplay`. Trying to directly bind a lambda also wouldn't work, because again `lambda` isn't exempted in `transient--redisplay`. (i.e. neither:

```elisp
(defun my-transient-scroll ()
  (interactive)
  (transient-scroll-up 1))
(define-key transient-map "x" 'my-transient-scroll)
(define-key transient-predicate-map [my-transient-scroll] 'transient--do-stay)
```
nor
```elisp
(define-key transient-map "x" (lambda () (interactive) (transient-scroll-up 1)))
```
works.)
